cluster-autoscaler/cloudprovider/huaweicloud/huaweicloud_service_manager.go

```diff
@@ -626,10 +627,36 @@ func (csm *cloudServiceManager) buildNodeFromTemplate(asgName string, template *
 
 	node.Labels = cloudprovider.JoinStringMaps(node.Labels, buildGenericLabels(template, nodeName))
 
+	node.Spec.Taints = extractTaintsFromTags(template.tags)
+
 	node.Status.Conditions = cloudprovider.BuildReadyConditions()
 	return &node, nil
 }
 
+func extractTaintsFromTags(tags map[string]string) []apiv1.Taint {
+	taints := make([]apiv1.Taint, 0)
+
+	for tagKey, tagValue := range tags {
+		// The tag value must be in the format <tag>:NoSchedule
+		r, _ := regexp.Compile("(.*):(?:NoSchedule|NoExecute|PreferNoSchedule)")
```

Comment:
Do we need to do this on the label? If yes, the user needs to adapt again. cc @RainbowMango